account_payment_term_partner_holiday/models/res_partner.py

```diff
@@ -16,31 +19,36 @@ class ResPartner(models.Model):
         auto_join=True
     )
 
+    def _get_valid_due_date(self, date):
+        if isinstance(date, str):
+            date = fields.Date.from_string(date)
+        is_date_in_holiday = self.is_date_in_holiday(date)
+        while is_date_in_holiday:
+            date = is_date_in_holiday[1]
+            is_date_in_holiday = self.is_date_in_holiday(date)
+        return date
+
     def is_date_in_holiday(self, date):
```

Comment:
As this is more than just a check, I would maybe rename the function (to `date_in_holiday`, for example) and add a docstring explaining that it should return the last day of vacation in case there is any for this period (check also my next comment for context)